### DISCUSSION TITLE
Importing CSV gives conversion error but log continues to be written

### DISCUSSION THREAD

**Discussion by sudo-asad** [firefly-iii/firefly-iii#8585] [Firefly III]
I am trying to use the latest data-importer with firefly-iii on a shared web host to import a CSV. The file has about 4000 entries. At the conversion stage, I get the " The conversion could not be started, or failed due to an error. Please check the log files. Sorry about this :( " error, but the log file in "storage/logs/" in the data-importer directory keeps getting written at about 1MB per 2 minutes or something. 

Edit: After waiting for more than half an hour or so after the error had appeared, the log file is no longer being written and these are the last few lines:

```
 local.DEBUG: Now in storeConversionStatus(conversion-mZjn9ScRdbeZ): conv_done  
 local.DEBUG: Messages: 0, warnings: 6, errors: 0  
 local.DEBUG: Set conversion as complete.  
```

I am assuming this means the conversion was successful but the webpage had already given that error and now there isn't a way for me to proceed. Any tips?

**Comment by JC5** [comment]
There are no tips for this really, except to import smaller batches or use the command line for large ones.

At some point, the front-end will timeout, but that doesn't mean the PHP thread has finished. So as you've noticed, you see an error about this, while the import continues in the background. There's not much I can do about that I'm afraid.

**Reply by sudo-asad** [reply]
Is there no way to increase the front-end timeout? I also noticed that after the error appears, the log fills up more slowly compared to before. 

I will try with command line if possible with my host, otherwise I'll use smaller batches.

**Reply by JC5** [reply]
> Is there no way to increase the front-end timeout? 

Yes, but that also means that you'll wait longer and longer to realize it actually crashed. The Docker container is set to 300 seconds I believe, you can set a self-managed installation to whatever you want.

**Comment by sudo-asad** [comment]
I tried both with CLI and short batches, but I get the following error in data-importer logs (after all the conversion process etc):

`Could not create transaction. Unexpected empty response from Firefly III. Check the logs.`

Then I enabled logging as described in the help pages for Firefly, but there isn't an "Error" per say in the logs. Just at the end there are a bunch of the following lines:

`[2024-02-25 20:22:13] local.DEBUG: Now in account search() `
repeated x number of times, and :

`[2024-02-25 20:22:24] local.DEBUG: GroupCollector: postFilterCollection has 0 filter(s) and 0 transaction(s). `
repeated x number of times. 

Also, I noticed there are a lot of 
```

[2024-02-25 20:20:21] local.DEBUG: convert()  
[2024-02-25 20:20:21] local.DEBUG: getCurrencyRate()  
[2024-02-25 20:20:21] local.DEBUG: getRate()  
```
lines. Do they imply some API/network operation to change or get the different rates? Is it possible to disable this at least for the import process as it is unnecessary for my case.

Thank you for all your help.

**Reply by JC5** [reply]
I see no errors, so I couldn't really say.
